routes/topics.js

```diff
@@ -16,6 +16,7 @@ router.get('/follow', isAuth, topicsController.getFollowTopic);
 router.get('/latest', validate(TopicValidation.latestPost), isAuth, topicsController.getLatestPost);
 router.put('/follow', isAuth, topicsController.putFollowTopic);
 router.put('/follow-v2', isAuth, topicsController.followTopicV2);
+router.put('/broadcast-message', isAuth, topicsController.broadcastMessage);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The new route `/broadcast-message` is a `PUT` request, which is typically used for updating resources. However, broadcasting a message seems more like an action rather than an update to a resource. Consider using a `POST` request instead if the broadcast action doesn't directly update a specific resource.

```diff
- router.put('/broadcast-message', isAuth, topicsController.broadcastMessage);
+ router.post('/broadcast-message', isAuth, topicsController.broadcastMessage);
```

Also, it's important to validate the input data before processing it. If there's a validation schema available for the broadcast message, it should be used here.

```diff
- router.post('/broadcast-message', isAuth, topicsController.broadcastMessage);
+ router.post('/broadcast-message', validate(BroadcastValidationSchema), isAuth, topicsController.broadcastMessage);
```
Please replace `BroadcastValidationSchema` with the actual validation schema for the broadcast message.


<!-- This is an auto-generated comment by OSS CodeRabbit -->